numpy/core/src/common/array_assign.c

```diff
@@ -64,19 +64,22 @@ broadcast_strides(int ndim, npy_intp const *shape,
     return 0;
 
 broadcast_error: {
-        PyObject *errmsg;
-
-        errmsg = PyUnicode_FromFormat("could not broadcast %s from shape ",
-                                strides_name);
-        PyUString_ConcatAndDel(&errmsg,
-                build_shape_string(strides_ndim, strides_shape));
-        PyUString_ConcatAndDel(&errmsg,
-                PyUnicode_FromString(" into shape "));
-        PyUString_ConcatAndDel(&errmsg,
-                build_shape_string(ndim, shape));
-        PyErr_SetObject(PyExc_ValueError, errmsg);
-        Py_DECREF(errmsg);
+        PyObject *shape1 = convert_shape_to_string(strides_ndim,
+                                                   strides_shape, "");
+        if (shape1 == NULL) {
+            return -1;
+        }
 
+        PyObject *shape2 = convert_shape_to_string(ndim, shape, "");
```

Comment:
As a follow-up to this collection of PRs, its very temping to remove the third argument from this function

Reply:
Good idea. It doesn't get used and is easy enough to add elsewhere if needed.